src/main/java/core/basesyntax/Zoo.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax;
+
+import core.basesyntax.abstractclasses.Animal;
+import core.basesyntax.interfaces.Flies;
+import core.basesyntax.interfaces.Swims;
+import core.basesyntax.interfaces.Walks;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Zoo {
+    private List<Flies> airHabitat = new ArrayList<>();
+    private List<Swims> waterHabitat = new ArrayList<>();
+    private List<Walks> landHabitat = new ArrayList<>();
+
+    public void addToWaterHabitat(Swims animal) {
+        waterHabitat.add(animal);
+    }
+
+    public void addToAirHabitat(Flies animal) {
+        airHabitat.add(animal);
+    }
+
+    public void addToLandHabitat(Walks animal) {
+        landHabitat.add(animal);
+    }
+
+    private void walkersWalk() {
+        for (Walks walker : landHabitat) {
+            walker.walk();
+        }
+    }
+
+    private void swimmersSwim() {
+        for (Swims swimmer : waterHabitat) {
+            swimmer.swim();
+        }
+    }
+
+    private void fliersFly() {
+        for (Flies flier : airHabitat) {
+            flier.fly();
+        }
+    }
+
+    private <E> void printAnimals(List<E> animals) {
```

Comment:
Спробуй щось таке:
і тоді не потрібно буде робити перевірку instanceof
```suggestion
    private void printAnimals(List<? extends Animal > animals) {
```

Reply:
Я все перпробував щоб позбавитись instanceof, але без нього не компілиться
Думаю що проблема в тому що я на вхід даю ArrayList інтерфейсів Swimming, Walking і Flying. Інтерфейси ніяк не пов'язані з Animal.
Хіба що записувати тваринок двічі - один раз у ArrayList вольєр, а другий - у ArrayList де їх можна накормити і вивести імена. Тільки це не правильно як на мене.